src/workerd/server/server.c++

```diff
@@ -1194,17 +1201,36 @@ public:
           };
 
           TimerChannel& timerChannel = service;
-          auto newActor = kj::refcounted<Worker::Actor>(
-              *service.worker, kj::mv(id), true, kj::mv(persistent),
-              className, kj::mv(makeStorage), lock,
+          auto newActorImpl = kj::heap<Worker::Actor::Impl>(
+              *service.worker, lock, kj::mv(id), true, kj::mv(persistent),
+              className, kj::mv(makeStorage),
               timerChannel, kj::refcounted<ActorObserver>());
 
-          return kj::HashMap<kj::String, kj::Own<Worker::Actor>>::Entry {
-            kj::mv(idStr), kj::mv(newActor)
+          return kj::HashMap<kj::String, LocalActor>::Entry {
+            kj::mv(idStr), LocalActor { .actor = kj::mv(newActorImpl) }
           };
-        }));
+        });
+
+        auto createActorRef = [&]() {
+          auto actorRef = kj::refcounted<Worker::Actor>(*localActor.actor);
```

Comment:
Isn't this equivalent to `kj::addRef()` or am I missing something?

Reply:
`*localActor.actor` is a `Worker::ActorImpl`, not a `Worker::Actor`. We allocated it via `kj::heap<Worker::ActorImpl>` and pass it to `Worker::Actor`s ctor. `kj::addRef(object)` requires the `object` to have been allocated via `refcounted<T>()` (i.e. its refcount must be >= 1), but we don't have a `Worker::Actor` yet, so we're doing that here now.